### ISSUE TITLE
Allow multiple blank node objects (mapping blank nodes)

### ISSUE THREAD

**Issue by simeonackermann** [RMLio/yarrrml-parser#101]
### Description
Thanks to #59 I'd expect the following should work:

```yaml
prefixes:
  ex: "http://example.com/"

mappings:
  person:
    sources:
      - ['data.json~jsonpath', '$.persons[*]']
    s: http://example.com/$(firstname)
    po:
      - [a, foaf:Person]
      - p: ex:friend
        o:
          value: $(friends[*].firstname)
          type: blank
```
Since with `type: literal` this will work, I guess it should also work with blank node objects? The error I get:

```
{"message":"Error while executing the rules.","log":"Error: Error while executing the rules.
    at /rmlmapper-webapi-js/node_modules/@rmlio/rmlmapper-java-wrapper/lib/wrapper.js:95:35
    at ChildProcess.exithandler (child_process.js:290:7)
    at ChildProcess.emit (events.js:193:13)
    at maybeClose (internal/child_process.js:999:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:266:5)"}
```

**Comment by pheyvaer** [comment]
Hi @simeonackermann 

What exactly do you want to achieve? Why does the object have a value if it needs to be a blank node?

**Comment by simeonackermann** [comment]
I have a dataset like this:

```json
{
    "persons": [
        {
            "firstname": "John",
            "lastname": "Doe"
        },
        {
            "firstname": "Jane",
            "lastname": "Smith",
            "friends": [
                {
                    "firstname": "Mr X"
                },
                {
                    "firstname": "Ms Y"
                }
            ]
        },
        {
            "firstname": "Sarah",
            "lastname": "Bladinck"
        }
    ]
}
```

With `type: literal` it creates the correct result:

```ttl
ex:John a foaf:Person.
ex:Jane a foaf:Person;
    ex:friends "Mr X", "Ms Y".
ex:Sarah a foaf:Person.
```

But this could also work blanknodes - not?  I wonder because it already works with targeting just ONE specific node: `value: $(friends[0].firstname)`. 
But maybe I'm totally wrong and should create a separate mapping for theses nodes (here friends) and link them each other.

**Comment by simeonackermann** [comment]
Hm, i wonder how to get blanknode objects and link them to my nodes ... like:

```
ex:John a foaf:Person.
ex:Jane a foaf:Person;
    ex:friends _:b1_mr_x , _:b2_ms_y.
ex:Sarah a foaf:Person.

_:b1_mr_x a foaf:Person .
_:b2_ms_y a foaf:Person .
```

**Comment by simeonackermann** [comment]
Ok, I try again, hope this will clarify my problem. I have the following JSON input data:
```json
{
    "persons": [
        {
            "firstname": "Monika",
            "files": [ { "name": "File A" } ]
        },
        {
            "firstname": "John",
            "files": [ { "name": "File B" } ]
        },
        {
            "firstname": "Ruben",
            "files": [ { "name": "File XYZ" } ]
        }
    ]
}
```
and want map it to this output data:
```ttl
example:Monika a foaf:Person;
    example:hasFile _:b1_0.
example:John a foaf:Person;
    example:hasFile _:b1_1.
example:Ruben a foaf:Person;
    example:hasFile _:b1_2.
_:b1_0 a example:File;
    rdfs:label "File A".
_:b1_1 a example:File;
    rdfs:label "File B".
_:b2_2 a example:File;
    rdfs:label "File XYZ".
```
I tried with a mapping function (see commented yarrrml, but how to map the blank nodes IRI?!), so I ended with `type: blank` objects, where I get an error from yarrrml-parser.
```yml
mappings:
  person:
    sources:
      - ['data.json~jsonpath', '$.persons[*]']
    s: example:$(firstname)_$(lastname)
    po:
      - [a, foaf:Person]
      - p: example:hasFile
        o: 
          value: $(files[*])
          type: blank
      #  o:
      #    - mapping: files
      #      condition:
      #        function: equal
      #        parameters:
      #          - [str1, $(???), s]
      #          - [str2, $(???), o]

  files:
    sources:
      - ['data.json~jsonpath', '$.persons[*].files[*]']
    s:
        type: blank
    po:
      - [a, example:File]
      - [rdfs:label, $(name)]
```

**Comment by simeonackermann** [comment]
Ok dude, finally I got it, sorry for the mess ... My simple solution was a `mapping` without condition ... :

```yaml
mappings:
  person:
    sources:
      - ['data.json~jsonpath', '$.persons[*]']
    s: ex:$(firstname)_$(lastname)
    po:
      - [a, foaf:Person]
      - p: exo:hasFile
        o: 
          mapping: files

  files:
    sources:
      - ['data.json~jsonpath', '$.persons[*].files[*]']
    s: null
    po:
      - [a, exo:File]
      - [rdfs:label, $(name)]
```

**Comment by simeonackermann** [comment]
Damn, I'm running into trubble again ...  it does'nt solved my problem, because now every person has every file ... 
Example:
```ttl
_:b6_0 a foaf:Person;
    ex:hasFile _:b6_1, _:b6_2, _:b6_3.
_:b6_4 a foaf:Person;
    ex:hasFile _:b6_1, _:b6_2, _:b6_3.
...
```

**Comment by simeonackermann** [comment]
Ok, it seems I need some join conditions, which are not jsonpath standard or use not RML-standard.
I tried with [RocketRML](https://github.com/semantifyit/RocketRML/issues/17) and here with jsonpath-plus its possible to select with `^` the parent object or alternatively to use the current `PATH~` to identify an object.

I noticed by the way, that the jsonpath standard to select the current element `$(.)` is not supported by yarrrml-parser..


Thanks anyway for this great lib!
